src/types/plugin-inputs.ts

```diff
@@ -0,0 +1,58 @@
+import { EmitterWebhookEvent as WebhookEvent, EmitterWebhookEventName as WebhookEventName } from "@octokit/webhooks";
+import { SupportedEventsU } from "./context";
+import { StaticDecode, Type as T } from "@sinclair/typebox";
+import { DiscriminatedUnionValidator } from "typebox-validators/discriminated";
+import { StandardValidator } from "typebox-validators";
+
+export interface PluginInputs<T extends WebhookEventName = SupportedEventsU> {
+  stateId: string;
+  eventName: T;
+  eventPayload: WebhookEvent<T>["payload"];
+  settings: PermitGenerationSettings;
+  authToken: string;
+  ref: string;
+}
+
+const permitRequestSchema = T.Union([
+  T.Object({
+    kind: T.Literal("ERC20"), // Discriminant key
+    type: T.Literal("ERC20"), // should remove this ideally
+    userId: T.Number(),
+    amount: T.Number({ minimum: 1 }),
+    evmNetworkId: T.Number(),
+    tokenAddress: T.String(),
+    issueNodeId: T.String(),
+  }),
+  T.Object({
+    kind: T.Literal("ERC721"), // Discriminant key
+    type: T.Literal("ERC721"), // should remove this ideally
```

Comment:
@gentlementlegen what do you think about this, I'd remove `type` and keep `kind` as TypeBox needs `kind` and this PR will be part of a larger refactoring of all infra that uses this so I think this is acceptable.